### PR TITLE
(fix) [SD-3599] Disabled users should not get error notifications

### PR DESCRIPTION
**Pull request by sivakuna-aap**

### CODE REVIEW

**Comment by mugurrus** [comment]
:+1:

File: superdesk/activity.py [hunk 1]
```diff
@@ -215,9 +215,10 @@ def notify_and_add_activity(activity_name, msg, resource=None, item=None, user_l
     add_activity(activity_name, msg=msg, resource=resource, item=item,
                  notify=[user.get("_id") for user in user_list] if user_list else None, **data)
     if user_list:
-        recipients = [user.get('email') for user in user_list if
-                      superdesk.get_resource_service('preferences').
-                      email_notification_is_enabled(preferences=user.get('user_preferences', {}))]
+        recipients = [user.get('email') for user in user_list
+                      if user.get('is_enabled', False) and user.get('is_active', False) and
```
Comment (mugurrus) [inline superdesk/activity.py]: A test for this would be helpful

**Comment by ioanpocol** [comment]
:+1:  (is needed a small test to prove that notifications are not sent anymore to disabled users)

File: superdesk/activity.py (hunk 1 above)
Comment (petrjasek) [inline superdesk/activity.py]: maybe this filtering should be done in `get_users_by_user_type` method?

Comment (sivakuna-aap) [inline superdesk/activity.py]: `get_users_by_user_type()` is being used in multiple places. Not sure if they need inactive users. That's the reason doing it in recipients.

**Comment by sivakuna-aap** [comment]
The fix has superdesk changes also. I will cover tests in superdesk PR.

File: superdesk/activity.py (hunk 1 above)
Comment (petrjasek) [inline superdesk/activity.py]: yep I was wondering about that

**Comment by petrjasek** [comment]
:+1:

**Pull request merged**